app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/processor/Fido2ProviderProcessorImpl.kt

```diff
@@ -124,10 +141,118 @@ class Fido2ProviderProcessorImpl(
         cancellationSignal: CancellationSignal,
         callback: OutcomeReceiver<BeginGetCredentialResponse, GetCredentialException>,
     ) {
-        // no-op: RFU
-        callback.onError(GetCredentialUnsupportedException())
+        cancellationSignal.setOnCancelListener {
+            callback.onError(GetCredentialCancellationException())
+            scope.cancel()
+        }
+
+        // If the user is not logged in, return an error.
+        val userState = authRepository.userStateFlow.value
+        if (userState == null) {
+            callback.onError(GetCredentialUnknownException("Active user is required."))
+            return
+        }
+
+        // Return an unlock action if the current account is locked.
+        if (userState.activeAccount.isVaultUnlocked.not()) {
+            val authenticationAction = AuthenticationAction(
+                title = context.getString(R.string.unlock),
+                pendingIntent = intentManager.createFido2UnlockPendingIntent(
+                    action = UNLOCK_ACCOUNT_INTENT,
+                    requestCode = requestCode.getAndIncrement(),
+                ),
+            )
+
+            callback.onResult(
+                BeginGetCredentialResponse(
+                    authenticationActions = listOf(authenticationAction),
+                ),
+            )
+            return
+        }
+
+        // Otherwise, find all matching credentials from the current vault.
+        try {
+            runBlocking {
+                val credentialEntries = getMatchingFido2CredentialEntries(
+                    userId = userState.activeUserId,
+                    request = request,
+                )
+
+                callback.onResult(
+                    BeginGetCredentialResponse(
+                        credentialEntries = credentialEntries,
+                    ),
+                )
+            }
+        } catch (e: GetCredentialUnknownException) {
+            callback.onError(e)
+        }
     }
 
+    @Throws
+    private suspend fun getMatchingFido2CredentialEntries(
+        userId: String,
+        request: BeginGetCredentialRequest,
+    ): List<CredentialEntry> =
+        request
+            .beginGetCredentialOptions
+            .flatMap { option ->
+                if (option is BeginGetPublicKeyCredentialOption) {
+                    val requestRpId = Json
+                        .decodeFromStringOrNull<PublicKeyCredentialRequestOptions>(
+                            option.requestJson,
+                        )
+                        ?.relayingPartyId
+                        ?: throw GetCredentialUnknownException("Invalid data.")
+
+                    vaultRepository
+                        .ciphersStateFlow
+                        .value
+                        .data
+                        ?.filter { it.isActiveWithFido2Credentials }
+                        ?.let {
+                            vaultSdkSource.decryptFido2CredentialAutofillViews(
```

Comment:
We can use SDK function `silentlyDiscoverCredentials(rpId)` to find matching credentials. It returns `List<Fido2CredentialAutofillView>`. We've not introduced it to `VaultSdkSource` yet, so you'll have to add it. The SDK call is...
```kotlin
getClient(userId)
  .platform()
  .fido2()
  .authenticator(userInterface, credentialStoreInterface)
  .silentlyDiscoverCredentials(rpId)
```

Reply:
Furthermore, I think we can add that function to our `VaultRepository` so the processor isn't dependent on `VaultSdkSource`.

Reply:
Awesome, will do, thanks!